FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift

```diff
@@ -343,3 +382,75 @@ extension FrontPageView: MarketsViewDelegate {
         delegate?.marketsView(marketsGridView, didSelectItemAtIndex: index)
     }
 }
+
+// MARK: - FrontPageShelfDatasource
+extension FrontPageView: FrontPageShelfViewDataSource {
+    public func frontPageShelfView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath, withItem item: AnyHashable) -> UICollectionViewCell? {
+        if let item = item as? RecentlyFavoritedViewmodel {
+            let cell = collectionView.dequeue(RecentlyFavoritedShelfCell.self, for: indexPath)
+            cell.configure(withModel: item)
+            cell.buttonAction = { [weak self] _, _ in
+                self?.removeFavoritedItem(item, atIndexPath: indexPath)
+            }
+            cell.datasource = self
+            cell.loadImage()
+            
+            return cell
+        } else if let item = item as? SavedSearchShelfViewModel {
+            let cell = collectionView.dequeue(SavedSearchShelfCell.self, for: indexPath)
+            cell.configure(withModel: item)
+            cell.imageDatasource = self
+            cell.loadImage()
+            return cell
+        }
+        return nil
+    }
+    
+    public func frontPageShelfView(cellClassesIn collectionView: UICollectionView) -> [UICollectionViewCell.Type] {
+        [RecentlyFavoritedShelfCell.self, SavedSearchShelfCell.self]
+    }
+    
+    public func datasource(forSection section: FrontPageShelfView.Section) -> [AnyHashable] {
+        guard let model = shelfViewModel else { return [] }
+        switch section {
+        case .savedSearch: return model.savedSearchItems
+        case .recentlyFavorited: return model.recentlyFavoritedItems
+        }
+    }
+    
+    public func removeFavoritedItem(_ item: AnyHashable, atIndexPath indexPath: IndexPath) {
+        guard
+            let viewModel = shelfViewModel,
+            let shelfView = frontPageShelfView
+        else { return }
+        viewModel.removeFavoritedItem(atIndex: indexPath.item)
+        shelfView.removeItem(item)
+    }
+}
+
+//MARK: - RemoteImageViewDataSource
+extension FrontPageView: RemoteImageViewDataSource {
+    public func remoteImageView(_ view: RemoteImageView, cachedImageWithPath imagePath: String, imageWidth: CGFloat) -> UIImage? {
+        return nil
+    }
+    
+    public func remoteImageView(_ view: RemoteImageView, loadImageWithPath imagePath: String, imageWidth: CGFloat, completion: @escaping ((UIImage?) -> Void)) {
```

Comment:
We already have a `CDNRemoteImageViewDataSource` in the app to handle image fetching. So you can add `remoteImageViewDataSource` to the init of this class, and use that one instead.